tests/shared/test_templates/template_test_nncf_tensor.py

```diff
@@ -0,0 +1,556 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import abstractmethod
+from typing import TypeVar
+
+import pytest
+
+from nncf.experimental.tensor import Tensor
+from nncf.experimental.tensor import TensorDataType
+from nncf.experimental.tensor import TensorDeviceType
+from nncf.experimental.tensor import functions
+
+TModel = TypeVar("TModel")
+TTensor = TypeVar("TTensor")
+
+
+# pylint: disable=too-many-public-methods
+class TemplateTestNNCFTensorOperators:
+    @staticmethod
+    @abstractmethod
+    def to_tensor(x: TTensor) -> TTensor:
+        pass
+
+    @pytest.mark.parametrize("operator", ("add", "sub", "mul", "pow", "truediv", "floordiv"))
+    def test_builtin_operator_tensor(self, operator):
+        tensor_a = self.to_tensor([1, 2])
+        tensor_b = self.to_tensor([22, 11])
+
+        nncf_tensor_a = Tensor(tensor_a)
+        nncf_tensor_b = Tensor(tensor_b)
+
+        if operator == "add":
```

Comment:
IMHO, this `if-elif` flow looks really weird.

Reply:
yes

Reply:
Discussed.